src/main.rs

```diff
@@ -203,48 +281,8 @@ fn run(arg_matches: ArgMatches) -> std::result::Result<(), String> {
 
     let mut runtime = TokioRuntime::new().unwrap();
 
-    // HTTP server for exporting Prometheus metrics
-    if let Some(addr) = arg_matches.value_of(args::HTTP_BIND_ADDR) {
-        let http_addr = addr
-            .parse()
-            .unwrap_or_else(|_| panic!("Unable to parse metrics address {}", addr));
-
-        let http_log = log.clone();
-
-        let http_server = hyper::Server::bind(&http_addr)
-            .serve(|| service_fn(metrics_service))
-            .map_err(move |e| error!(http_log, "HTTP Server error: {}", e));
-
-        info!(log, "Starting Prometheus {} exporter.", app::NAME; "address" => &http_addr);
-        let _http_thread = runtime.spawn(http_server);
-    }
-
-    let mut server = Server::with_root(home_dir)
-        .greeting("Welcome to unFTP")
-        .authenticator(make_auth(&arg_matches))
-        .passive_ports(49152..65535)
-        .with_metrics();
-
-    // Setup FTPS
-    server = match (
-        arg_matches.value_of(args::FTPS_CERTS_FILE),
-        arg_matches.value_of(args::FTPS_KEY_FILE),
-    ) {
-        (Some(certs_file), Some(key_file)) => {
-            info!(log, "FTPS enabled");
-            server.certs(certs_file, key_file)
-        }
-        (Some(_), None) | (None, Some(_)) => {
-            warn!(
-                log,
-                "Need to set both {} and {}. FTPS still disabled.", "ftps-certs-file", "ftps-key-file"
-            );
-            server
-        }
-        _ => server,
-    };
-
-    let _ftp_thread = runtime.spawn(server.listener(&addr));
+    runtime = start_http(&log, &arg_matches, runtime);
```

Comment:
can we maybe pass mutable reference only? this looks quite hackish (but maybe there's a reason?)

Reply:
We can if you think that is better.

Reply:
See if you think the latest commit makes it better.